src/tiered/tiered_handle.c

```diff
@@ -8,63 +8,488 @@
+    __wt_scr_free(session, &tmp);
+    return (ret);
+}
+
+/*
+ * __tiered_create_tier_tree --
+ *     Create a tier name for a tiered table.
+ */
+static int
+__tiered_create_tier_tree(WT_SESSION_IMPL *session, WT_TIERED *tiered)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    const char *cfg[4] = {NULL, NULL, NULL, NULL};
+    const char *config, *name;
+
+    conn = S2C(session);
+    config = name = NULL;
+    WT_RET(__wt_scr_alloc(session, 0, &tmp));
+
+    /* Create tier:example for the new tiered tree. */
+    WT_ERR(__wt_tiered_name(session, &tiered->iface, 0, WT_TIERED_NAME_SHARED, &name));
+    cfg[0] = WT_CONFIG_BASE(session, tier_meta);
+    WT_ASSERT(session, conn->tiered_prefix != NULL);
+    WT_ASSERT(session, conn->bstorage != NULL);
+    WT_ERR(__wt_buf_fmt(
+      session, tmp, ",bucket=%s,bucket_prefix=%s", conn->bstorage->bucket, conn->tiered_prefix));
+    cfg[2] = tmp->data;
+    WT_ERR(__wt_config_merge(session, cfg, NULL, &config));
+    /* Set up a tier:example metadata for the first time. */
+    __wt_verbose(
+      session, WT_VERB_TIERED, "TIER_SWITCH: schema create TIERED_TREE: %s : %s", name, config);
+    WT_ERR(__wt_schema_create(session, name, config));
+    WT_ASSERT(session, tiered->tier_names[WT_TIERED_INDEX_READONLY] == NULL);
+    tiered->tier_names[WT_TIERED_INDEX_READONLY] = name;
+
+    if (0)
+err:
+        /* Only free on error. */
+        __wt_free(session, name);
+    __wt_free(session, config);
+    __wt_scr_free(session, &tmp);
+    return (ret);
+}
+
+/*
+ * __tiered_update_dhandles --
+ *     Update the dhandle list for a tiered structure after object switching.
+ */
+static int
+__tiered_update_dhandles(WT_SESSION_IMPL *session, WT_TIERED *tiered)
+{
+    WT_DECL_RET;
+    uint32_t i;
+
+    /* Now get the dhandle and add it to the array. */
+    for (i = 0; i < WT_TIERED_MAX_TIERS; ++i) {
+        /*
+         * If we have a tiered dhandle we can either skip if it is the same name or we decrement the
+         * old one and get a new one for the new name.
+         */
+        if (tiered->tiers[i] != NULL) {
+            WT_ASSERT(session, tiered->tier_names[i] != NULL);
+            if (strcmp(tiered->tiers[i]->name, tiered->tier_names[i]) == 0)
+                continue;
+            else
+                (void)__wt_atomic_subi32(&tiered->tiers[i]->session_inuse, 1);
+        }
+        if (tiered->tier_names[i] == NULL)
+            continue;
+        __wt_verbose(
+          session, WT_VERB_TIERED, "UPDATE_DH: Get dhandle for %s", tiered->tier_names[i]);
+        WT_ERR(__wt_session_get_dhandle(session, tiered->tier_names[i], NULL, NULL, 0));
+        __wt_verbose(
+          session, WT_VERB_TIERED, "UPDATE_DH: DONE Get dhandle for %s", tiered->tier_names[i]);
+        (void)__wt_atomic_addi32(&session->dhandle->session_inuse, 1);
+        tiered->tiers[i] = session->dhandle;
+        WT_ERR(__wt_session_release_dhandle(session));
```

Comment:
`__tiered_init_tiers` has similar code to get the dhandle.  Consider a shared function/macro?